### DISCUSSION TITLE
Does `satnum` matter? (satrec.sgp4init argument)

### DISCUSSION THREAD

**Discussion by CDWimmer** [skyfielders/python-skyfield#943] [Q&A] [answered]
Hello!

I think not, from a poke at the source, but just to check:- does the 3rd argument to `satrec.sgp4init()` actually do anything of consequence? 

I'm not sure what to pass to this from the newer GP element standards given the old 5-digit IDs are defunct and the newer OMM data from CelesTrak provides an alphanumeric `OBJECT_ID`s instead. I *think* it'll just deal with any `int` or `str` entered here but the lack of typing in these libraries makes it unobvious, and the name `satnum`/"satellite number" (or even `satn` deeper within sgp4) is a bit of a misnomer if allowed to be used this way 

I'm currently tempted to enter `1` in everything for now or just use the `OBJECT_ID` string; recommendations welcome!

I suppose this could well spin off a conversation about direct support for the post-TLE data formats, which given it looks like they're the direction that sources like CelesTrak want to prefer in future, should perhaps be something to consider?

As an aside, if there is a stubs lib to add some typing to sgp4 and/or skyfield I'd love to know, I've spoiled myself as of late by throwing type hints at everything and am feeling lost without them 😂

---
References for newer standards: [https://public.ccsds.org/Pubs/502x0b3e1.pdf](https://public.ccsds.org/Pubs/502x0b3e1.pdf) and [https://celestrak.org/NORAD/documentation/gp-data-formats.php](https://celestrak.org/NORAD/documentation/gp-data-formats.php)

**Comment by brandon-rhodes** [comment]
Well — that depends on what kinds of effects you think are 'of consequence', versus effects that are not. Could you explain what your threshold is?

**Comment by CDWimmer** [comment]
Hi, thank you for the quick reply! I just mean will it affect any calculations, i.e. Ground position coordinates,  or is it just a display/`__repr__`(?) thing? I didn't have long to look at it last night, apologies.

**Reply by brandon-rhodes** [reply, marked as answer]
Thanks for clarifying! So the answer is no: the satellite catalog number does not affect its coordinates or motion. It simply changes what is printed out if you print the satellite as a whole, or the satellite catalog number specifically.

**Reply by CDWimmer** [reply]
Perfect thank you, I'll just feed the OBJECT_ID into it then :) Do you think it would be on the cards to support the OMM data types in future?

**Reply by brandon-rhodes** [reply]
Happily, the OMM format can already be loaded into an `sgp4` Satrec in Python, and then used in Skyfield. I just haven't had time yet to sit down and write up an example for the Skyfield documentation. Here's the issue tracking that request, in case you want to watch for when it's finally complete:

https://github.com/skyfielders/python-skyfield/issues/763

**Reply by CDWimmer** [reply]
Lovely, I have been following the guide at the bottom of the one documentation page to do exactly that. Apologies for missing the issue, I'll give it a follow :)

**Reply by CDWimmer** [reply]
I may look at making a pull req for a `parse_json(file)` or something like that in sgp4 - do you have a stance on the use of type annotations? 

edit: ah, obviously no annotation due to <3.3 and <2.7 support! Maybe a community type stubs library would be in order :P
